subprojects/delivery/nupokati/src/main/kotlin/com/avito/android/provider/CdBuildConfigTransformer.kt

```diff
@@ -0,0 +1,37 @@
+package com.avito.android.provider
+
+import com.avito.android.model.CdBuildConfig
+import com.google.gson.FieldNamingPolicy
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import org.gradle.api.Transformer
+import org.gradle.api.file.ProjectLayout
+import java.io.File
+
+internal class CdBuildConfigTransformer(
+    private val rootProjectLayout: ProjectLayout,
+    private val validator: CdBuildConfigValidator,
+) : Transformer<CdBuildConfig, String> {
+
+    override fun transform(configFilePath: String): CdBuildConfig {
+        val configFile = rootProjectLayout.projectDirectory.file(configFilePath).asFile
+
+        return deserializeToCdBuildConfig(configFile).also {
+            // todo test on failed validation
+            validator.validate(it)
+        }
+    }
+
+    private fun deserializeToCdBuildConfig(configFile: File): CdBuildConfig {
+        require(configFile.exists()) { "Can't find cd config file in $configFile" }
+        return uploadCdGson.fromJson(configFile.reader(), CdBuildConfig::class.java)
+    }
+}
+
+internal val uploadCdGson: Gson by lazy {
+    GsonBuilder()
+        .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
```

Comment:
Does it mean that we have values without `@SerializedName`?
I'd prefer to describe it explicitly and don't rely on any naming policy. It's too easy to break.

Reply:
Yep, i just copied what we have already
It's really less obvious, but i see a hidden profit: it will be a little easier to migrate from gson that way